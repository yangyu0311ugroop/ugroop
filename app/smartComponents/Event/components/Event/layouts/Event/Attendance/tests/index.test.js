/**
 * Created by stephenkarpinskyj on 16/11/18.
 */

import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import { EventAttendance } from '..';

describe('<EventAttendance />', () => {
  let wrapper;
  let instance;

  const makeProps = () => ({
    dailyAttendance: true,
  });

  beforeEach(() => {
    wrapper = shallow(<EventAttendance {...makeProps()} />);
    instance = wrapper.instance();
  });

  it('exists', () => {
    expect(EventAttendance).toBeDefined();
  });

  describe('#renderField()', () => {
    it('still matches snapshot', () => {
      expect(
        toJSON(shallow(<div>{instance.renderField()}</div>)),
      ).toMatchSnapshot();
    });
  });

  describe('#renderEditable()', () => {
    it('still matches snapshot', () => {
      expect(
        toJSON(shallow(<div>{instance.renderEditable()}</div>)),
      ).toMatchSnapshot();
    });
  });

  describe('#render()', () => {
    it('not explodes', () => {
      expect(wrapper).toHaveLength(1);
    });

    it('still matches snapshot', () => {
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });
});
