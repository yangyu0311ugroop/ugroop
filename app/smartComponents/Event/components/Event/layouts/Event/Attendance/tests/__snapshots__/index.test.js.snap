// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EventAttendance /> #render() still matches snapshot 1`] = `""`;

exports[`<EventAttendance /> #renderEditable() still matches snapshot 1`] = `
<div>
  <WithStyles(GridItem)>
    <WithStyles(GridContainer)
      alignItems="baseline"
    >
      <WithStyles(GridItem)>
        <EventHeading
          Typography={[Function]}
        >
          <FormattedMessage
            defaultMessage="Record Attendance:"
            id="app.Event.layouts.Event.Attendance.editableHeadingLabel"
          />
        </EventHeading>
      </WithStyles(GridItem)>
      <ConnectFunction
        dailyAttendance={true}
        readOnly={false}
        variant={null}
      />
      <WithStyles(GridItem)>
        ·
      </WithStyles(GridItem)>
      <ConnectFunction
        dailyAttendance={true}
        readOnly={false}
        variant={null}
      />
      <WithStyles(GridItem)>
        ·
      </WithStyles(GridItem)>
      <ConnectFunction
        dailyAttendance={true}
        readOnly={false}
        variant={null}
      />
    </WithStyles(GridContainer)>
  </WithStyles(GridItem)>
</div>
`;

exports[`<EventAttendance /> #renderField() still matches snapshot 1`] = `
<div>
  <WithStyles(Section)
    ExpansionPanelProps={
      Object {
        "defaultExpanded": false,
      }
    }
    title={
      <FormattedMessage
        defaultMessage="Record Attendance"
        id="app.Event.layouts.Event.Attendance.sectionLabel"
      />
    }
  >
    <WithStyles(GridContainer)>
      <ConnectFunction
        dailyAttendance={true}
        readOnly={false}
        variant={null}
      />
      <ConnectFunction
        dailyAttendance={true}
        readOnly={false}
        variant={null}
      />
      <ConnectFunction
        dailyAttendance={true}
        readOnly={false}
        variant={null}
      />
    </WithStyles(GridContainer)>
  </WithStyles(Section)>
</div>
`;
